### PR TITLE
Learning deep metrics

### PR DESCRIPTION
**Pull request by jodoin**

### CODE REVIEW

File: article/_posts/2019-07-17-HDML.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+![](/article/images/hdml/sc05.jpg)
+
+The system has also two more losses $$J_m$$ and $$J_{syn}$$, a triplet or contrastive loss (they tested both) applied to $$z$$ and $$\tilde{z}$$.
+
+![](/article/images/hdml/sc06.jpg)
+
+
+# Results
+
+Their method... works! on a variaty of datasets.
```
Comment (jhlegarreta) [inline article/_posts/2019-07-17-HDML.md]: variety

File: article/_posts/2019-07-17-HDML.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+![](/article/images/hdml/sc01.jpg)
+![](/article/images/hdml/sc02.jpg)
+
+
+# Methods
+
+The overall system is illustrated in Fig.3.
+![](/article/images/hdml/sc03.jpg)
+
+It starts with a GoogleNet which projects the input image to a vector space $$Y$$.  The follows an autoencoder which learns a Euclidean latent space $$Z$$.  The synthetic "harder" example is then generated as follows
```
Comment (jhlegarreta) [inline article/_posts/2019-07-17-HDML.md]: _The follows an autoencoder (...)_ -> _Then an autoencoder follows (...)_? or _It is followed by an autoencoder (...)_ ?

File: article/_posts/2019-07-17-HDML.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    title:   "Hardness-Aware Deep Metric Learning"
+    venue:   "CVPR 2019"
+pdf: "https://arxiv.org/pdf/1903.05503.pdf"
+---
+
+
+# Introduction
+The goal of this paper is to learn deep metrics.  A deep metric is a function which computes the similarity between two input signals (here images).  This may be done with a contrastive loss or a triplet loss which both consider a set of 3 (triplet loss) or 4 (contrastive loss) positive and negative (read similar and different) images to teach a neural net to project those points into a low-dimensional Euclidean space in which similar pointis are close and dissimilar points are far. 
+
+The authors argue that the more a neural net is shown difficiult sets of points (negatives images with similar content or positive images with different content) the better that system will be.  In that perspective, the author proposed a novel **hardness-ware strategy**.  This paper sits on three main ideas namely:
```
Comment (jhlegarreta) [inline article/_posts/2019-07-17-HDML.md]: _In that perspective, the authors (...)_ just to be consistent.

File: article/_posts/2019-07-17-HDML.md (hunk 1 above)
Comment (jodoin) [reply article/_posts/2019-07-17-HDML.md]: done

File: article/_posts/2019-07-17-HDML.md (hunk 2 above)
Comment (jodoin) [reply article/_posts/2019-07-17-HDML.md]: done

**Pull request merged**